st2actions/st2actions/runners/ssh/paramiko_ssh.py

```diff
@@ -144,6 +149,9 @@ def connect(self):
                  '_username': self.username, '_timeout': self.timeout}
         self.logger.debug('Connecting to server', extra=extra)
 
+        if self.bastion_host and self._bastion_connect():
+            conninfo['sock'] = self.bastion_socket
```

Comment:
`ParamikoSSH` attempts to open a socket to the target SSH server if a `sock` parameter isn't given. If it's given a `sock` parameter, it won't attempt to connect and will use that socket blindly.

https://github.com/paramiko/paramiko/blob/master/paramiko/client.py#L279

The intent behind the code is to connect to the bastion host, open a `Channel` to the host behind the bastion host, and pass that to the 'real' ssh client getting used.

http://docs.paramiko.org/en/1.15/api/client.html#paramiko.client.SSHClient.connect

It's the equivalent of
```
ssh -L 1234:realhost:22 stanley@bastionhost
ssh stanley@localhost -p 1234
```
Instead of
```
ssh stanley@realhost
```
Because the original connection is occurring via a tunnelled ssh connection, environment variables are being preserved

Any pointers on how those integration tests would be setup/how I can run them locally?